doc/build-helpers/fixed-point-arguments.chapter.md

```diff
@@ -0,0 +1,89 @@
+# Fixed-point arguments of build helpers {#chap-build-helpers-finalAttrs}
+
+As mentioned in the beginning of this part, `stdenv.mkDerivation` could alternatively accept a fixed-point function. The input of such function, typically named `finalAttrs`, is expected to be the final state of the attribute set.
+A build helper like this is said to accept **fixed-point arguments**.
+
+As [`stdenv.mkDerivation`'s support of fixed-point arguments](#mkderivation-recursive-attributes) was first included in the Nixpkgs release 22.05, `mkDerivation`-style build helpers don't always come with such support yet.
+
+## Defining a build helper with `lib.extendMkDerivation` {#sec-build-helper-extendMkDerivation}
+
+The Nixpkgs library function [`lib.customisation.extendMkDerivation`](#function-library-lib.customisation.extendMkDerivation) can be used to define a build helper supporting fixed-point arguments from an existing one with such support. It can also bring fixed-point argument support to existing builders as a wrapper around the original definition.
```

Comment:
`lib.extendMkDerivation` doesn't take a conventional build helper and modify it. We modify the Nix expression that defines the build helper to make it support fixed-point arguments.

However, we don't need to implement from scratch the logic to consume `finalAttrs: { ... }` as `stdenv.mkDerivation` does, nor do we need to pass all the arguments to the base build helper and ship the build-helper-specific implementation (as an overlay) through `.overrideAttrs` (what #194475 is doing). Instead, we pass the base build helper and an overlay-like implementation to `lib.extendMkDerivation`, and it will return the build helper we want.